example/qubole_presto_api_example.py

```diff
@@ -1,68 +1,79 @@
-
 """
 This is the sample code used for submitting a Presto query (PrestoCommand) and getting the result back to local file.
 Similar way can be followed for HiveCommand etc.
 """
-
-import sys
-import string
-from ConfigParser import SafeConfigParser
+ 
+import logging, sys, string, time
+from tempfile import NamedTemporaryFile
+from configparser import SafeConfigParser
 from qds_sdk.qubole import Qubole
-from qds_sdk.commands import *
+from qds_sdk.commands import PrestoCommand
 import boto
-import time
-
+import pandas as pd
 
-# Used for generating file name to download the result
-def get_random_filename(size=10):
-    return "/tmp/result_" + str(int(time.time())) + ".tsv"
+# Setting up the logger
+logging.basicConfig(stream=sys.stdout,
+                    format='[%(asctime)s] [%(filename)s] [%(levelname)s] %(message)s',
+                    level=logging.WARN)
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
 
-# Returning content from the file 
-def get_content(filename):
-    with open(filename, 'r') as content_file:
-        content = content_file.read()
-    return content
 
-# Executing given query
-def execute_query(query):
-    if query is None or query == "":
-        return None
-    cmd = PrestoCommand.create(query=query)
-    query_id = str(cmd.id)
-    print "Starting Command with id: " + query_id + "\nProgress: =>",
+#######################
+# PRIVATE FUNCTIONS
+#######################
+def _wait_for_cmd(cmd, update=True):
     while not PrestoCommand.is_done(cmd.status):
```

Comment:
won't this get stuck for forever here, if `update` is `False`?

Reply:
yes that's true. I've just changed it to include a timeout

Reply:
thanks